### ISSUE TITLE
float division by zero error at BayesianNetwork.bake()

### ISSUE THREAD

**Issue by fareez-ahamed** [jmschrei/pomegranate#13]
Hi

I'm trying to create a simple Bayesian Network with two states, but I'm confused with the way **ConditionalProbabilityTable** works.

```python
# The hardware works correcty 99 out 100 times
hardware = DiscreteDistribution( { 'w': 0.99, 'nw': 0.01 } )


# The editor works if the hardware works
editor = ConditionalProbabilityTable(
		[['w', 'w', 1],
		 ['nw', 'w', 0],
     	 ['w', 'nw', 0],
     	 ['nw', 'nw', 1]],[hardware])

s1 = State( hardware, name="hardware" )
s2 = State( editor,name="editor")

# Create the Bayesian network object with a useful name
network = BayesianNetwork( "Freds LISP Problem" )


network.add_nodes([s1, s2])

network.add_edge(s1, s2)
network.bake()
```

This runs without error, whereas if I modify the CPT to have different keys, it ends up in float by division error at `network.bake()`

```python
# The editor works if the hardware works
editor = ConditionalProbabilityTable(
		[['w', 'work', 1],
		 ['nw', 'work', 0],
     	 ['w', 'notwork', 0],
     	 ['nw', 'notwork', 1]],[hardware])
```

Two different states can take up different sets of values, but the CPT doesnt seems to accept it

**Comment by jmschrei** [comment]
Issue referenced from #11 . This is an issue with pomegranate I will fix today.

**Comment by yeskarthik** [comment]
Am facing this issue as well.

**Comment by jmschrei** [comment]
Sorry about the delay in fixing this. The bug was difficult to track down, as it ended up not being in any of the math. The bug was that ConditionalProbabilityTable.keys() was returning the keys of the final parent, not the variable it was representing. This is consistent with my examples, which all have the same sets of keys, making this issue irrelevant.

I added the simplified titanic example as a unit test. The new version is on both github and PyPi. Closing this issue, reopen if it doesn't work for you!
